File: backend/src/core/domain/venta.py
class Venta:
    def __init__(self, receta_nombre: str, cantidad: int, fecha: str, id: str = None):
        self.id = id
        self.receta_nombre = receta_nombre
        self.cantidad = cantidad
        self.fecha = fecha
        
    def to_dict(self):
        return {
            "id": self.id,
            "receta_nombre": self.receta_nombre,
            "cantidad": self.cantidad,
            "fecha": self.fecha
        }
        
    @classmethod
    def from_dict(cls, data: dict):
        return cls(
            receta_nombre=data["receta_nombre"],
            cantidad=data["cantidad"],
            fecha=data.get("fecha"),
            id=data.get("id")
        )

File: internal/core/domain/venta.go
package domain

import "strings"

// Venta es una unidad (o varias) de una receta vendida en una fecha.
type Venta struct {
	ID           string
	RecetaNombre string
	Cantidad     int
	Fecha        Date
}

// Validar comprueba las invariantes mínimas.
func (v Venta) Validar() error {
	if strings.TrimSpace(v.RecetaNombre) == "" {
		return &ValidationError{Campo: "receta_nombre", Msg: "el nombre de la receta es obligatorio"}
	}
	if v.Cantidad <= 0 {
		return &ValidationError{Campo: "cantidad", Msg: "la cantidad debe ser mayor que cero"}
	}
	if v.Fecha.IsZero() {
		return &ValidationError{Campo: "fecha", Msg: "la fecha es obligatoria"}
	}
	return nil
}
